provider/testdata/src/lib.rs

```diff
@@ -149,7 +137,10 @@ use icu_provider_adapters::fallback::LocaleFallbackProvider;
 /// The return type of this method is not considered stable, mirroring the unstable trait
 /// bounds of the constructors. For matching versions of `icu` and `icu_testdata`, however,
 /// these are guaranteed to match.
-#[cfg(any(feature = "all_features_hack", not(feature = "bin")))] // allow regenerating testdata even if databake doesn't compile
+#[cfg(any(
+    feature = "internal_all_features_hack",
+    not(feature = "internal_ignore_baked")
+))] // allow accessing metadata even if databake doesn't compile
```

Comment:
Suggestion: Instead of doing gymnastics with negative features, can you just put the baked data behind a (default) feature, which the bin crate can disable with `default-features = false`?

Reply:
That would be a breaking change for using `icu_testdata` with `default-features = false`.

Reply:
Do we know whether this is a high-impact change in ICU4X components? I think most people who use no-default-features add back a few components like `icu_decimal` or `icu_normalizer`, so those features could imply the baked feature